src/client/java/net/neoforged/neoforge/client/event/ConfigureGpuDeviceEvent.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.event;
+
+import net.neoforged.bus.api.Event;
+import net.neoforged.fml.event.IModBusEvent;
+import net.neoforged.neoforge.client.neo3d.GpuDeviceFeatures;
+import net.neoforged.neoforge.client.neo3d.GpuDeviceProperties;
+
+public class ConfigureGpuDeviceEvent extends Event implements IModBusEvent {
+    private final GpuDeviceProperties deviceProperties;
+    private final GpuDeviceFeatures availableFeatures;
+    private final GpuDeviceFeatures enabledFeatures;
+
+    public ConfigureGpuDeviceEvent(GpuDeviceProperties deviceProperties, GpuDeviceFeatures availableFeatures, GpuDeviceFeatures enabledFeatures) {
+        this.deviceProperties = deviceProperties;
+        this.availableFeatures = availableFeatures;
+        this.enabledFeatures = enabledFeatures;
+    }
+
+    public GpuDeviceProperties getDeviceProperties() {
+        return deviceProperties;
+    }
+
+    public GpuDeviceFeatures getAvailableFeatures() {
+        return availableFeatures.clone();
+    }
+
+    public GpuDeviceFeatures getEnabledFeatures() {
+        return enabledFeatures.clone();
+    }
+
+    public void requireFeatures(GpuDeviceFeatures requiredFeatures) {
```

Comment:
I don't think this is a good way to do it. I would much rather have a simple `boolean` getter and setter for each feature. Maybe on an interface rather than the event so we could make it abstract.

Reply:
I'm not opposed to a different solution, how exactly would you have your proposed alternative work?

Reply:
We'd need to go for the intermediate for binary compat. The same 'struct' here is also re-used later to report what is actively enabled, would that be its own thing, or?

Reply:
No idea, it would be worth experimenting with different approaches. Presumably the interface can be reused for the getters, and the setters can be in the event.

Reply:
@Technici4n is this resolved now? also the related comment for GpuDeviceFeatures.